### PR TITLE
feat: flip USE_MECH_ANALYTICS default to on

### PR DESCRIPTION
**Pull request by DIvyaNautiyal07**
## Summary

Flips the `USE_MECH_ANALYTICS` flag default from opt-in (`=== 'true'`) to opt-out (`!== 'false'`) so mech-analytics is the primary source for the per-request mech feed and `senderTotal` on the predict page. Rollback to the subgraph is a one-liner env-var flip.

## Prerequisites (all landed)

- **mech-analytics PR #36** (valory-xyz/mech-analytics#36) — undelivered on-chain requests now land as shell rows so `COUNT(*)` matches on-chain reality. Deployed and verified live: `~292k` historical undelivered rows backfilled, live scoring flowing.
- **Placeholder repair** — one-shot rewrote 204 historical trader-tool rows that had `prompt='[offchain request]'` (mech-side bug that shipped before valory-xyz/mech#478). Real question text pulled back from the marketplace subgraph.
- **Parity re-run** (Aug 1-25 window) after both fixes landed:
  - `row_parity.missing_in_analytics`: **225 → 21** (matches the 204 repair count)
  - `senderTotal`: 302,465 = 302,465 (PASS, delta 0 on omenstrat, delta 1 on polystrat within ±5 tolerance)
  - Residual 21 missing are 2 mech-side attribution edge cases + ~19 non-trader Safes using trader tools; none affect user-visible metrics

## Test plan

- [ ] Deploy to staging, `USE_MECH_ANALYTICS` unset → reads should flow through mech-analytics; verify ROI + senderTotal render correctly on `/predict`
- [ ] Verify rollback path: set `USE_MECH_ANALYTICS=false` on the staging env, redeploy, confirm reads route back to the subgraph
- [ ] Watch for `MechAnalyticsError` in the server logs during the first 24-48h of production traffic

## Rollback

Set `USE_MECH_ANALYTICS=false` in the deployment env. No revert needed.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: common-util/api/predict/mech-analytics.ts [hunk 1]
```diff
@@ -7,8 +7,12 @@ import { QMR_MAX_AGE_DAYS } from 'common-util/constants';
...
  */
 
-/** One switch that turns on all mech-analytics read paths. */
-export const USE_MECH_ANALYTICS = process.env.USE_MECH_ANALYTICS === 'true';
+/** One switch that turns on all mech-analytics read paths. Default ON
+ * after the mech-analytics cutover (undelivered rows admitted as shell
+ * shape in mech-analytics PR #36 + placeholder repair against the
+ * marketplace subgraph); explicit ``USE_MECH_ANALYTICS=false`` still
+ * routes reads back through the subgraph as a rollback lever. */
+export const USE_MECH_ANALYTICS = process.env.USE_MECH_ANALYTICS !== 'false';
```
Comment (Tanya-atatakai) [inline common-util/api/predict/mech-analytics.ts]: `docs/mech-analytics-migration.md` still says "**Status:** `USE_MECH_ANALYTICS` is off everywhere" and "Default: off" — and this comment points readers to that doc. the doc header even asks to be updated when the flag changes :D can we update the status + default lines in this PR?

Comment (Tanya-atatakai) [inline common-util/api/predict/mech-analytics.ts]: question: previews / local dev that never set `MECH_ANALYTICS_URL` used to silently take the subgraph path; now they hit "MECH_ANALYTICS_URL not set" on every refresh run by default (roi-distribution soft-fails and keeps stale data, tool-accuracy drops senders with a console.error). is that intended for non-prod envs, or should a missing URL route back to the subgraph? lmk if all envs already have the URL set

Comment (Tanya-atatakai) [inline common-util/api/predict/mech-analytics.ts]: tiny, up to you: the rollback lever requires the exact string `false` — someone typing `FALSE`/`0` during an incident silently won't roll back and won't notice. wonder if we should accept a case-insensitive `false`? same for townhall-kpis #151 if yes

File: .env.example [hunk 2]
```diff
@@ -85,4 +85,4 @@ SOLANA_RPC=__URL__
 
 # mech-analytics API (off-chain migration, server-only) — see docs/mech-analytics-migration.md
 MECH_ANALYTICS_URL=__URL__
-USE_MECH_ANALYTICS=false
+USE_MECH_ANALYTICS=true
```
Comment (Tanya-atatakai) [inline .env.example]: tiny: `=true` is now a no-op (unset already means on). maybe document the rollback lever instead, wdyt:
```suggestion
USE_MECH_ANALYTICS= # unset = on; set to false to roll reads back to the subgraph
```

**Review by Tanya-atatakai — DISMISSED** [review]
approving with caveats — the flip itself is right; please sync the migration doc with the new default (comment below), the rest is optional

File: common-util/api/predict/mech-analytics.ts (hunk 1 above)
Comment (DIvyaNautiyal07) [reply common-util/api/predict/mech-analytics.ts]: Good catch, fixed. Rewrote the status header, the flag/default sentence in the intro, and ticked off steps 1-3 of the rollout section. Step 4 (remove flag + subgraph paths) stays as the follow-up after the watch window.

Comment (DIvyaNautiyal07) [reply common-util/api/predict/mech-analytics.ts]: Not sure prod always has it, so I added the guard defensively. USE_MECH_ANALYTICS now also checks !!process.env.MECH_ANALYTICS_URL, so an env without the URL falls back to the subgraph instead of hitting the fetch error. Previews and local dev get the safe default with no config change on their end.

Comment (DIvyaNautiyal07) [reply common-util/api/predict/mech-analytics.ts]: Done, made it case-insensitive on both PRs. FALSE / False / faLSE all trigger the rollback now.

File: .env.example (hunk 2 above)
Comment (DIvyaNautiyal07) [reply .env.example]: Applied your suggestion, thanks.

**Review by OjusWiZard — DISMISSED** [review]

**Review by OjusWiZard — APPROVED** [review]

**Review by Tanya-atatakai — APPROVED** [review]
